pkg/k8s/reconcileutil/update_from_env.go

```diff
@@ -19,10 +19,12 @@ func UpdateFromEnv(ctx context.Context, client client.Client, obj runtime.Object
 
 	for i := range specs {
 		val, ok := os.LookupEnv(specs[i].Env)
-		if ok {
-			changed = true
-			*specs[i].Target = val
+		if !ok || *specs[i].Target == "" {
```

Comment:
> Overriding via environment variables now only updates empty values in the spec.

It looks to me like this does the opposite. It skips the update if the value is empty.

Also, might be clearer if this condition is placed before `LookupEnv` since it is independent of that.

Reply:
Thanks for spotting this! That's what I get for not testing :sweat_smile: